src/Cake.Common/Tools/OpenCover/OpenCoverRegisterOption.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.OpenCover
+{
+    /// <summary>
+    /// Represents the register-options:
+    ///  - empty for admin-registry-access
+    ///  - "user" for user-registry-access
+    ///  - path for non-registry-dll dll
+    /// </summary>
+    public class OpenCoverRegisterOption
+    {
+        private string value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenCoverRegisterOption"/> class.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        public OpenCoverRegisterOption(string value)
+        {
+            this.value = value;
+        }
+
+        /// <summary>
+        /// Converts to string.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="System.String" /> that represents this instance.
+        /// </returns>
+        public override string ToString()
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                return string.Empty;
+            }
+
+            return $":{value}";
+        }
+
+        /// <summary>
+        /// Gets the register-option representing the "user"-mode.
+        /// (this will translate to -register:user)
+        /// this
+        /// </summary>
+        public static OpenCoverRegisterOption User => new OpenCoverRegisterOption("user");
```

Comment:
This will create a new object each time which is unnecessary. I think you can use
`public static OpenCoverRegisterOption User { get; } = new OpenCoverRegisterOption("user");`